Initial backend prototype

KluvaDa opened the pull request:
Initial prototype of the backend. User authentication is currently mocked. The functionality of the API is just storing and retrieving key-value pairs of data. Testing is implemented and checks are passed locally (with a local MySQL instance).

KluvaDa committed:
update README.md

The first readme update that serves as a plan and backbone for the first contribution and development.

KluvaDa committed:
Untested - intial commit with the development of the basic functionality. OAuth is not implemented yet. API supports saving and retrieving data in a field specified using any key per user.

KluvaDa committed:
Untested - Implement comprehensive test suite and documentation

- Create database test utilities with dual-database safety architecture
- Implement pytest configuration with async support and fixtures
- Add comprehensive database tests: CRUD operations, user isolation, edge cases
- Create API tests with concurrency and load testing
- Implement automatic pytest detection for test database selection
- Add database name validation to prevent production accidents
- Create clean test structure with logical organization
- Update documentation to reflect implemented functionality
- Add key-value storage functionality to README
- Fix documentation typos and outdated references

KluvaDa committed:
BROKEN - testing in progress

Replaced DB_NAME and DB_NAME_TEST from .env and am using separate .env for test dev and prod. The tests will only run if APP_ENV=test in the .env and error out otherwise.

KluvaDa committed:
Debug and fix errors in test_database.py; test_api.py remains broken

- Addressed and resolved errors in test_database.py through config and test utility updates.
- Refactored database config and test setup to improve reliability.
- test_api.py is still failing and requires further investigation.
- Updated .gitignore and removed obsolete db_inspect.log.

KluvaDa committed:
Debug and fix test isolation for key-value API and DB; all tests now pass

- Fixed test database isolation by ensuring all cleanup operations in fixtures are properly committed, preventing state leakage between tests.
- Refactored test fixtures for clarity and correctness:
  - Split session management, DB cleanup, and FastAPI dependency overrides into separate fixtures.
  - Ensured each test and API request gets a fresh, isolated session.
- Verified that all API and repository tests pass reliably, both in isolation and as a suite.
- Confirmed that the codebase currently uses mock authentication and only exposes the user key-value getter/setter endpoints.
- Documented and resolved issues with missing commits, fixture scoping, and session concurrency.
- The codebase is now ready for further feature development with a robust, reliable test foundation.

KluvaDa committed:
Improved documentation, docstrings, comments, readme

KluvaDa committed:
minor

alisiddiqi reviewed (commented):
Looks good

KluvaDa committed:
Updated readme installation and prerequisites instructions.

KluvaDa committed:
Added uvicorn.run() into app.main

KluvaDa committed:
Updated to new versions of packages in pyproject.toml and updated code for new version of httpx

KluvaDa committed:
Work in progress.

Trying to encapsulate everything in docker for both production and development

KluvaDa committed:
Removed .vscode/settings.json from repository. (It should never have been committed)

KluvaDa committed:
Removed .vscode/settings.json from repository. (It should never have been committed)

KluvaDa committed:
Untested - implemented the docker containerisation

sirinoks commented on README.md:
````diff
@@ -1 +1,124 @@
+   ```
+3. **Run the app:**
+   ```bash
+   python -m app.main
````
As mentioned in discord discussion, `poetry run python -m app.main` instead

KluvaDa committed:
Big rework of the repo to support a docker setup. I have overengineered since I didn't realise that vs code offers the container setup with tools and additional packages installed with its configuration files. This made my whole dev and dev-db setup unnecessarily complicated. I have not written the documentation or checked it - it was auto-generated by the ai and is probably missing the mark. I will simplify to rely on vs code functionality next

KluvaDa committed:
Documentation is still missing and outdated!

Continuing to work on the vs code container configuration.

Changed git settings to keep all files with linux style line endings LF.

Still relying on dev section in dockerfile, just adding a layer of devcontainer.json for further setup afterwards.

Some fixes with permissions and bind mounts

KluvaDa committed:
minor - example files for configuration, added a basic .vscode configuration, modified the .example usage so that the default example configuration is also committed to the repo as a default but is gitignored for further user customisation.

KluvaDa committed:
Work in progress - documentation and docker and vs code setup

Commit:
Work in progress. documentation is unwritten

Trying to set up a robust development docker container

Commit:
Work in progress - fixing docker setup inside dev container

Commit:
Fixed pytest vs code integration and docker fully shutting down when vs code is closed.

Commit:
Fixed the run testing and debug configuration

Commit:
Changed .gitignore and structure of how dev files are committed. They now exist in dev_example and are untracked everywhere else.

Commit:
Deleting file that was never meant to exist

Commit:
Added cleanup to the test task

Commit:
Added script for offline container use

Commit:
Updated documentation and some minor changes. Updating the configuration files too. We still have issues with warnings during testing

Commit:
Fixed warnings in testing.
Had to remove PYTHONOPTIMIZE so that we use ssertions for testing, and fixed permissions. Also some readibility improvements.

KluvaDa committed:
Merge pull request #2 from questoflife/dev-daniel-docker

Added docker containerization

KluvaDa commented:
I have now added docker containerization and fully documented. It now includes containers for production and testing, as well as the whole development/debugging environment and vs code integration configurations.

You should be able to follow the documentation to set it up and run the pytest locally.

@sirinoks @alisiddiqi could you have a look, see if you can follow the instructions, run it, and pass the tests locally before approving the pull request?

After we merge it, I will start working on running it in the cloud. We can then also start designing the frontend, at least for integration tests at this point to see if the setup works. And setting up discord oauth will be the next thing we'll need to do too, but I'm hoping we can start having smaller contributions that always stem from github issues at this point.

KluvaDa committed:
Improved the documentation about the dev_example directory

KluvaDa committed:
Updated documentation

KluvaDa committed:
Updated documentation and comments in file for docker and docker-compose version requirements. Also added the dockerfile version

KluvaDa committed:
Updated permissions for docker in docker commands and updated the documentation for clarity

KluvaDa committed:
Added TLS to database. No custom CA certificate support though.

KluvaDa committed:
Fixing TLS ssh for database

KluvaDa committed:
Added TLS disable flag for local development and testing

KluvaDa committed:
Removed database creation and deletion that requires admin access

KluvaDa committed:
added a cors middleware layer and allowed origins

KluvaDa commented:
I am developing actively too much. I stopped the pull request for now and will open a new one once it is ready to be deployed.
